### PR TITLE
SASS-353 - Introduced 'Add to list' pattern in the interest account su…

### PR DESCRIPTION
**Pull request by Dula-J**
…mmary page

### CODE REVIEW

File: test/views/interest/InterestAccountsViewSpec.scala [hunk 1]
```diff
@@ -167,9 +186,9 @@ class InterestAccountsViewSpec extends ViewTest {
 
       "there are two accounts passed in" which {
 
-        lazy val result = view(taxYear, Seq(
+        lazy val result = view(untaxedYesNoForm, taxYear, Seq(
           InterestAccountModel(None, "Bank of UK", 9000.01, Some("qwerty")),
-          InterestAccountModel(Some("azerty"), "Bank of EU", 1234.56)
+          InterestAccountModel(Some("azerty"), "Bank of EU", 1234.56) // This is a prior submission
```
Comment (robert-webb-hmrc) [inline test/views/interest/InterestAccountsViewSpec.scala:191]: Do we need the comment text here?

**Review by robert-webb-hmrc — CHANGES_REQUESTED** [review]

File: test/views/interest/InterestAccountsViewSpec.scala (hunk 1 above)
Comment (Dula-J) [reply test/views/interest/InterestAccountsViewSpec.scala:191]: Will remove it

File: app/views/interest/InterestAccountsView.scala.html [hunk 2]
```diff
@@ -78,51 +81,65 @@
...
+    <li class="hmrc-add-to-a-list__contents">
+        <span class="hmrc-add-to-a-list__identifier hmrc-add-to-a-list__identifier--light">
+            @account.accountName
+        </span>
+        @if(account.id.nonEmpty) {
+            <span class="hmrc-add-to-a-list__change"></span>
+        }
+        <span class="hmrc-add-to-a-list__change">
+            <a class="govuk-link" href="@changeLink">
+                <span aria-hidden="true">@messages("common.change")</span>
```
[the remark this answers is not in the document]
Comment (BenCosfordHMRC) [reply app/views/interest/InterestAccountsView.scala.html:94]: I think keep the span as aria hidden but include the "change" & "remove" in hidden text

[the remark this answers is not in the document]
Comment (Dula-J) [reply app/views/interest/InterestAccountsView.scala.html:94]: I don't have a screen reader set up on my machine and no audio output, but i remember @joeyymoore mentioning that adding the verb into the hidden span will result in "Remove Remove Sydney Russell" being read out. Can someone try this out?

[the remark this answers is not in the document]
Comment (robert-webb-hmrc) [reply app/views/interest/InterestAccountsView.scala.html:94]: That's why I raised this originally, i used voiceover to verify. Anything labelled aria hidden is not read out by the screen reader, which is why we need the verb in the description as well.

[the remark this answers is not in the document]
Comment (Dula-J) [reply app/views/interest/InterestAccountsView.scala.html:94]: Got you, I'll make the change

File: test/views/interest/InterestAccountsViewSpec.scala [hunk 3]
```diff
@@ -32,12 +33,23 @@ class InterestAccountsViewSpec extends ViewTest {
   val TAXED = "taxed"
   val UNTAXED = "untaxed"
 
-  val accountRow: Int => String = rowNumber => s".govuk-summary-list__row:nth-child($rowNumber)"
-  val accountRowName: Int => String = rowNumber => accountRow(rowNumber) + " > dd:nth-child(1)"
-  val accountRowChange: Int => String = rowNumber => accountRow(rowNumber) + " > dd:nth-child(2) > a"
-  val accountRowRemove: Int => String = rowNumber => accountRow(rowNumber) + " > dd:nth-child(3)"
+  private val untaxedYesNoForm =
+    YesNoForm.yesNoForm("Select yes if you received taxed interest from the UK").bind(Map("value" -> "true"))
+  private val taxedYesNoForm =
```
Comment (robert-webb-hmrc) [inline test/views/interest/InterestAccountsViewSpec.scala:38]: Are these vals the wrong way around?

**Review by robert-webb-hmrc — CHANGES_REQUESTED** [review]
Just the one test to check. Can we also address these warns as well please?

Screenshot 2021-03-11 at 16 59 42

Screenshot 2021-03-11 at 16 58 56

File: test/views/interest/InterestAccountsViewSpec.scala (hunk 3 above)
Comment (Dula-J) [reply test/views/interest/InterestAccountsViewSpec.scala]: yes, good spot

**Review by robert-webb-hmrc — APPROVED** [review]
Changes approved, just waiting on journey tests to be merged

**Pull request merged**